icssc/AntAlmanac
# Add schedule notes

## Summary
- Users can add a note for each of their schedules. They can also edit, save, and load their notes.
- Users can write notes while they are adding/editing a schedule.
- Schedule notes will be displayed under their added classes; the user can also edit the note directly in the box.

Task list:
- [x] Allow user to write/edit a note while they create or edit a schedule.
- [x] Add a box under the added classes that displays the current schedule's notes.
- [x] Allow schedule notes to be saved and loaded (waiting for schedule structure rewrite). For old schedules, make sure each schedule's note is just an empty string.
- [x] Add a way to edit a schedule note by clicking somewhere in the note box.

## Test Plan
- For a schedule with no notes, make sure the note box displays a message that says there is no note.
- Create a new schedule with notes. Make sure the note is displayed properly in the box.
- Edit a schedule that already has a note, and make sure the note box changes.
- Create multiple schedules with notes. Switch between these schedules and make sure the displayed notes are correct.
- When you delete a schedule, make sure the notes are still displayed correctly for the remaining schedules.
- Test various notes. Make sure newlines are processes and the max character limit is enforced.
- Save the schedules and see if the notes are still there when loaded back.
- Load older schedules that were saved with no schedule notes, then try editing notes for these schedules.

## Issues

Closes #413
